dhis2/dhis2-core
# fix: SqlView JSON and XML response with pager [DHIS2-9813]

### Summary
Reverts the first attempt to fix the issue in #11234 as it was lacking the root object with pager level.

New solution adds the `GridResponse` as root object using the existing `Pager`.
For JSON Jackson takes care of the serialisation. As Jackson support XML was removed the `GridUtils` got extended so the can handle a `GridResponse` and write the additional fields to XML.

### Manual Testing 
* login as `system` user (or any other with `ALL` authority)
* create a new SQL query view
* check JSON output with pager `/api/29/sqlViews/{uid}/data?paging=true`
* check JSON output without pager `/api/29/sqlViews/{uid}/data?paging=false`
* check XML output with pager `/api/29/sqlViews/{uid}/data.xml?paging=true` 
* check XML output without pager `/api/29/sqlViews/{uid}/data.xml?paging=false`

check here means to check it looks like a valid JSON/XML document and compare the structure to that returned by older versions.They should have the same structure.

### Example Output
JSON (shortened for main structure, `headers` and `rows` usually not empty)
```json
{
  "pager": {
    "page": 1,
    "pageCount": 1,
    "total": 22,
    "pageSize": 50
  },
  "listGrid": {
    "headerWidth": 15,
    "width": 15,
    "title": "test",
    "height": 22,
    "headers": [],
    "rows": []
  }
}
```

XML (also shortened)
```xml
<metadata xmlns="http://dhis2.org/schema/dxf/2.0">
<pager>
<page>5</page>
<pageCount>11</pageCount>
<total>22</total>
<pageSize>2</pageSize>
</pager>
<listGrid title="test" width="15" height="2">
<headers>
<header name="categoryid" column="categoryid" type="java.lang.String" hidden="false" meta="false"/>

</headers>
<rows>
<row>

</row>
</rows>
</listGrid>
</metadata>
```
